### ISSUE TITLE
Is there any polling mechanism in wmediumd.c ？

### ISSUE THREAD

**Issue by zijinchen** [ramonfontes/wmediumd_802154#1]
Hello, I have checked your wmediumd.c file and found the CSMA code, but I did not find the polling code, I wonder if there is any?Thank you very much!

**Comment by ramonfontes** [comment]
Hi,

I haven't finished the coding yet.

**Comment by zijinchen** [comment]
@ramonfontes 
Thank you very much for your reply! Because I see that the mininet-wifi project uses wmediumd on your version, and I want to confirm that you mean that there is no polling function in the current code, right? Do you plan to add it later? Or is it not necessary? Thank you very much!

**Comment by ramonfontes** [comment]
Are you refering to this repo https://github.com/ramonfontes/wmediumd?

**Comment by zijinchen** [comment]
@ramonfontes 
What I see is this：https://github.com/ramonfontes/wmediumd_802154
I haven't noticed the one you are referring to. Could you briefly tell me the difference between the two projects?

**Comment by ramonfontes** [comment]
Wmediumd was designed for 802.11 and I was trying to extend it to 805.15.4

**Comment by zijinchen** [comment]
@ramonfontes 
Great job! I know that there is CSMA in 802.11, but I am not sure whether there is a polling mechanism in 802.11.Is there any reflection in your two projects?thank you！

**Comment by ramonfontes** [comment]
It has a backoff time https://github.com/ramonfontes/wmediumd/blob/mininet-wifi/wmediumd/wmediumd.c#L384 that I'm not pretty sure that is working correctly.  I would say that today's biggest problem with wmediumd is the separation of the wireless medium. For example, data traffic exchanged between 2 nodes associated with one access point directly interferes with simultaneous traffic between 2 other nodes associated with another access point.

**Comment by zijinchen** [comment]
@ramonfontes 
You are absolutely right. This is the second question I want to ask, because I see that the four configurations you mentioned don't take into account the interference between signals, for example, different signals on the same channel? I don't know if this description is the same as what you said about the biggest problem?

**Comment by ramonfontes** [comment]
Yes, they don't take into account the interference between two signals on the same channel too. However, even if you add two nodes associated with one AP and other two nodes associated to another AP (far from the first three nodes), it doesn't work well. I started to work with this 15 days ago with https://github.com/ramonfontes/wmediumd/tree/parallel, but I'm not sure whether this is the correct direction.

**Comment by zijinchen** [comment]
@ramonfontes 
However, the interference between multiple signals and multiple aps are more in line with the actual situation. What you mean is that even if two aps (with some nodes) are far away from each other and there is no signal interference, the effect is not very good, right?So far only a single AP can be satisfied?

**Comment by ramonfontes** [comment]
You can see the data being exchanged through the 2 APs. The problem is: if you try Iperf between the first 2 clients and throughput is 10Mbp/s; and simultaneously do the same between the other 2 clients, the throughput will be split between all nodes, resulting in lower throughput.

**Comment by zijinchen** [comment]
Sorry, I am a little confused. According to your description, the throughput of the first two clients is set to 10Mbp/s, and the throughput of the last two clients is also set to 10Mbp/s, so the throughput will decrease. Is the throughput of the whole system fixed?The more AP, the less throughput per allocation?

**Comment by ramonfontes** [comment]
The throughput decreases after starting the second instance of iperf, which is being used with nodes that aren't associated with the same AP.

**Comment by zijinchen** [comment]
Aha!  isn't that the "distraction" we want? Is it strange that you set the distance between the two AP's far? There should be no interference in theory, but there is interference in practice, Right?

**Comment by ramonfontes** [comment]
No, we don't want this when there is no signal overlap.

**Comment by zijinchen** [comment]
The first sentence was just a joke, I know what you mean, the signals should not interfere with each other without overlapping, but in fact the experiment you did, the two far away from the AP caused interference, now there is a problem, right?

**Comment by ramonfontes** [comment]
Okay! :) well, I suppose there are two big problems now. ;)

By the way, are you trying to fix the problem mentioned by you?

**Comment by zijinchen** [comment]
Yes, we are also committed to solving this big problem. We hope to communicate with you more. Could you please keep one email address?

**Comment by ramonfontes** [comment]
Yes, sure. Please let me know if you need help or discuss more about this.  There are many strands about mininet-wifi and I have not been able to handle it all. :|  

We could have done all this discussion on \wmediumd

ramonreisfontes@gmail.com
